### PR TITLE
RAG Endpoint and moved embeddings

### PR DESCRIPTION
**Pull request by rachaelcodes**
## Context

We're writing an endpoint for RAG-enabled chat. Unlike the previous version of this, we want to be able to restrict the files being searched to those the user has access to.

Given that we have been discussing a model where the user-file connection is stored in the Django database, my assumption is that the Django app would provide a list of files that could be referenced (which in turn could be customised by the user themselves) along with the ChatHistory.

## Changes proposed in this pull request

This is an incomplete draft - I'm being pulled into some other projects, so I'm sharing my work so far in case anyone else has time to pick it up.

### Next steps

* fix the vector_store instantiation
* limit the files searched
* testing

## Relevant links
https://technologyprogramme.atlassian.net/browse/REDBOX-105

## Things to check

- [ ] I have added any new ENV vars in all deployed environments
- [ ] I have tested any code added or changed
- [ ] I have run integration tests https://github.com/i-dot-ai/redbox-copilot/actions/runs/8832554814

### CODE REVIEW

File: core_api/src/routes/chat.py [hunk 1]
```diff
@@ -25,12 +34,31 @@
...
+# TODO: do we want to be able to set hybrid to True depending on Elastic subscription levels?
+# see: https://github.com/i-dot-ai/redbox-copilot-streamlit/blob/3d197e76e6d42cfe0b70f66bb374c899b74754b4/streamlit_app/utils.py#L316C17-L316C45
+hybrid = False
+
+# TODO: fix this - I don't think it's correctly implemented, as we're getting an error when making a request:
+# raise HTTP_EXCEPTIONS.get(meta.status, ApiError)(
+# elasticsearch.NotFoundError: NotFoundError(404, 'status_exception', '[file] is not an inference service model or a deployed ml model')
+vector_store = ElasticsearchStore(
+    es_connection=es,
+    index_name="redbox-data",
```
Comment (gecBurton) [inline core_api/src/routes/chat.py:57]:
```suggestion
    index_name="redbox-data-chunk",
```
maybe should be `file`? not sure tbh

File: core_api/src/routes/chat.py [hunk 2]
```diff
@@ -25,12 +34,31 @@
...
+    embedding=embedding_function,
+    strategy=ApproxRetrievalStrategy(hybrid=hybrid, query_model_id="file"),
```
Comment (gecBurton) [inline core_api/src/routes/chat.py:59]:
```suggestion
    strategy=ApproxRetrievalStrategy(hybrid=hybrid, query_model_id=env.embedding_model),
```
something like this?

**Comment by gecBurton** [comment]
@lmwilkigov to simplify this PR a little i have raised the docker changes in their own PR here https://github.com/i-dot-ai/redbox-copilot/pull/267 pls could you take a look and then rebase/merge back into yours?

File: .github/workflows/python-app.yml [hunk 3]
```diff
@@ -52,7 +52,6 @@ jobs:
         docker compose up -d --wait elasticsearch
 
         poetry install --no-root --no-ansi --with dev --without ai,api,ingester
-        poetry run download-model --model_name paraphrase-albert-small-v2
```
Comment (gecBurton) [inline .github/workflows/python-app.yml:55]: is this the issue?

Comment (lmwilkigov) [reply .github/workflows/python-app.yml:55]: AH

**Pull request closed without merging**